### PR TITLE
OpenMP + HIP build

### PR DESCRIPTION
**Pull request by rhornung67**
# Summary

- This PR fixes issue with _OPENMP not defined in rocm device compiler pass.
- It also fixes example so it builds when openmp and hip are both enabled.

### CODE REVIEW

**Review by davidbeckingsale — APPROVED** [review]

**Comment by davidbeckingsale** [comment]
Is OpenMP enabled for HIP builds in GitLab CI?

**Comment by rhornung67** [comment]
@davidbeckingsale not yet. I am collecting CI improvements in an issue that I want to address after the release. That is one of them.

**Comment by MrBurmark** [comment]
If this is only to avoid the `#error` then I'm ok with this.

**Comment by MrBurmark** [comment]
I'm concerned about things like this breaking in a hip + openmp build https://github.com/LLNL/RAJA/blob/develop/include/RAJA/policy/hip/reduce.hpp#L623.

**Comment by rhornung67** [comment]
It is my understanding that _OPENMP *is not* defined in the device compile pass (based on the watercooler discussion today). So it seems that we need to pay close attention to that so we don't break things. Am I missing something?

**Comment by trws** [comment]
It shouldn't be defined in device code when compiling for hip. That check actually looks broken to me, it would cause ABI breakage to compile some parts of a code with `-fopenmp` and others without with that the way it is.  I'd err on the side of wasting the memory regardless so we can avoid that minefield personally.

**Comment by rhornung67** [comment]
@trws I've got a few changes coming that will fix the concerns folks have. Stay tuned.

File: cmake/SetupPackages.cmake [hunk 1]
```diff
@@ -82,7 +82,7 @@ if (RAJA_ENABLE_HIP)
   endif()
 
   if (RAJA_ENABLE_EXTERNAL_ROCPRIM)
-    find_package(RocPRIM)
+    include(cmake/thirdparty/FindRocPRIM.cmake)
```
Comment (rhornung67) [inline cmake/SetupPackages.cmake:85]: @davidbeckingsale are you OK with this change? It eliminates a CMake warning about rocprim package name mismatch with newer CMake versions.

**Comment by rhornung67** [comment]
@trws we have several of these checks in the code: #if defined(RAJA_ENABLE_OPENMP) && defined(_OPENMP). @MrBurmark and I have been debating whether they should only check RAJA_ENABLE_OPENMP and remove the _OPENMP check. What do you think?

**Comment by trws** [comment]
I think it depends a little on what they are for.  If it impacts the structure of data or interfaces, then I think it should only check `ENABLED`, so that we don't end up introducing ABI or API breaks between files, but for some things `ACTIVE` (or whether we're compiling *this* file for OpenMP) is a reasonable check, like for whether some templates need to be accessible in the current TU.  Does that make sense?

**Comment by rhornung67** [comment]
@trws I think that makes sense. I will iterate with @MrBurmark.

**Comment by rhornung67** [comment]
@trws when you have a few minutes, please look this over again. We think the OpenMP conditional compilation checks are good now.

**Comment by trws** [comment]
They look reasonable to me, thanks @rhornung67!

**Comment by rhornung67** [comment]
@trws thanks for checking. I will add a HIP + OMP build pipeline to our CI.

**Pull request merged**